### PR TITLE
:recycle: (autocomplete) cleaning up state updates

### PR DESCRIPTION
**Pull request by MatissJanis**
Cleaning up the autocomplete state updates by removing the `updater` and moving to react state hooks.

This is mostly a copy&paste, but I also fixed a few small issues and improved things a bit.. apologies for the big PR, but making this smaller would be very tricky.

### CODE REVIEW

**Comment by MatissJanis** [comment]
Ok, this is now open for review.

File: packages/desktop-client/src/components/util/GenericInput.js [hunk 1]
```diff
@@ -45,23 +45,21 @@ export default function GenericInput({
     case 'id':
       switch (field) {
         case 'payee':
-          if (payees.length > 0) {
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/util/GenericInput.js:48]: It did not show the payee field before if there were no payees. So you just get.. nothing. Which is not great UX.

Now we will show an empty payee field. And if you start typing in it - you will get the "create payee" option. Which is slightly better.

File: packages/desktop-client/src/components/autocomplete/Autocomplete.js [hunk 2]
```diff
@@ -624,7 +505,6 @@ export function MultiAutocomplete({
         item => !selectedItems.includes(getItemId(item)),
       )}
       onSelect={onAddItem}
-      clearAfterSelect
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/autocomplete/Autocomplete.js:627]: No point in this prop if it is `true` everywhere..

**Review by j-f1 — APPROVED** [review]

File: packages/desktop-client/src/components/autocomplete/Autocomplete.js [hunk 3]
```diff
@@ -405,100 +100,198 @@ function defaultShouldSaveFromKey(e) {
...
-      onSelect={updater(onSelect)}
+      onSelect={(item, { inputValue }) => {
+        setSelectedItem(item);
+        setHighlightedIndex(null);
+        setIsOpen(false);
+
+        if (onSelect) {
+          // I AM NOT PROUD OF THIS OK??
+          // This WHOLE FILE is a mess anyway
+          // OK SIT DOWN AND I WILL EXPLAIN
```
Comment (j-f1) [inline packages/desktop-client/src/components/autocomplete/Autocomplete.js:181]: Nice keeping this :)

**Pull request merged**